lib/devise/token_authenticatable/model.rb

```diff
@@ -36,7 +36,13 @@ module Models
     #
     module TokenAuthenticatable
       extend ActiveSupport::Concern
-
+      
+      Devise::Models.config(self, :should_reset_authentication_token, :should_ensure_authentication_token)
+      included do
+        before_save :reset_authentication_token if (should_reset_authentication_token ||= false)
```

Comment:
If I am not mistaken Rails enables you to use conditional callbacks:
```
before_save :reset_authentication_token, if: :should_reset_authentication_token?
```
Please, use this one.